azurerm/internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -394,6 +418,13 @@ func resourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta
 		}
 	}
 
+	if conn.TrafficSelectorPolicies != nil {
```

Comment:
we need to ensure that `traffic_selector_policy` is set at all times - as such can we remove this? the flatten function should return an empty list if this is nil/empty - which should then show a diff

```suggestion
```